### PR TITLE
*: add `SESSION_ALIAS` column in processlist table and expensive query log

### LINKED ISSUE #46889: add `SESSION_ALIAS` column in processlist table and expensive query log

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #46889

### What is changed and how it works?

A new column `SESSION_ALIAS` will be added to table `information_schema.PROCESSLIST` and `information_schema.CLUSTER_PROCESSLIST` to display session_alias 

```
TiDB root@127.0.0.1:test> set @@session.tidb_session_alias="alias123";
Query OK, 0 rows affected
Time: 0.001s
TiDB root@127.0.0.1:test> select * from information_schema.PROCESSLIST\G
***************************[ 1. row ]***************************
ID             | 2097154
USER           | root
HOST           | 127.0.0.1:56620
DB             | test
COMMAND        | Query
TIME           | 0
STATE          | autocommit
INFO           | select * from information_schema.PROCESSLIST
DIGEST         | 4b5e7cdd5d3ed84d6c1a6d56403a3d512554b534313caf296268abdec1c9ea99
MEM            | 0
DISK           | 0
TxnStart       |
RESOURCE_GROUP | default
SESSION_ALIAS  | alias123
```

Also add `session_alias` field to expensive query log:

```
[2023/09/12 11:59:40.902 +08:00] [WARN] [expensivequery.go:145] [expensive_query] [cost_time=60.932767459s] [conn=2097154] [user=root] [database=test] [txn_start_ts=0] [mem_max="0 Bytes (0 Bytes)"] [sql="select sleep(100)"] [session_alias=alias123]
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
add `SESSION_ALIAS` column in processlist table and expensive query log
```

### CODE REVIEW

File: util/processinfo.go [hunk 1]
```diff
@@ -51,16 +51,17 @@ type ProcessInfo struct {
 	DiskTracker           *disk.Tracker
 	StatsInfo             func(interface{}) map[string]uint64
 	RuntimeStatsColl      *execdetails.RuntimeStatsColl
-	DB                    string
+	User                  string
 	Digest                string
 	Host                  string
-	User                  string
+	DB                    string
```
Comment (lcwangchao) [inline util/processinfo.go:57]: That change is fixed by `fieldalignment` tool, or else the bazel lint will fail

**Review by hawkingrei — APPROVED** [review]

File: infoschema/tables.go [hunk 2]
```diff
@@ -835,6 +835,7 @@ var tableProcesslistCols = []columnInfo{
 	{name: "DISK", tp: mysql.TypeLonglong, size: 21, flag: mysql.UnsignedFlag},
 	{name: "TxnStart", tp: mysql.TypeVarchar, size: 64, flag: mysql.NotNullFlag, deflt: ""},
 	{name: "RESOURCE_GROUP", tp: mysql.TypeVarchar, size: resourcegroup.MaxGroupNameLength, flag: mysql.NotNullFlag, deflt: ""},
+	{name: "SESSION_ALIAS", tp: mysql.TypeString, size: 64, flag: mysql.NotNullFlag, deflt: ""},
```
Comment (bb7133) [inline infoschema/tables.go:838]: `mysql.TypeString` will be generated as `binary`, please change it to `TypeVarchar`.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**